airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -254,15 +211,29 @@ public static void runServer(final Set<ContainerRequestFilter> requestFilters,
 
     if (airbyteDatabaseVersion.isPresent() && AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())) {
       LOGGER.info("Starting server...");
-      new ServerApp(configRepository, jobPersistence, configs, requestFilters, responseFilters).start();
+
+      final JobTracker jobTracker = new JobTracker(configRepository, jobPersistence);
+      final WorkflowServiceStubs temporalService = TemporalUtils.createTemporalService(configs.getTemporalHost());
+      final TemporalClient temporalClient = TemporalClient.production(configs.getTemporalHost(), configs.getWorkspaceRoot());
+      final SchedulerJobClient schedulerJobClient = new DefaultSchedulerJobClient(jobPersistence, new DefaultJobCreator(jobPersistence));
+      final DefaultSynchronousSchedulerClient syncSchedulerClient = new DefaultSynchronousSchedulerClient(temporalClient, jobTracker);
+      final SpecCachingSynchronousSchedulerClient cachingSchedulerClient = new SpecCachingSynchronousSchedulerClient(syncSchedulerClient);
+
+      return apiFactory.create(
+          schedulerJobClient,
+          cachingSchedulerClient,
+          temporalService,
+          configRepository,
+          jobPersistence,
+          configs);
     } else {
       LOGGER.info("Start serving version mismatch errors. Automatic migration either failed or didn't run");
-      new VersionMismatchServer(airbyteVersion, airbyteDatabaseVersion.get(), PORT).start();
+      return new VersionMismatchServer(airbyteVersion, airbyteDatabaseVersion.get(), PORT);
     }
   }
 
   public static void main(String[] args) throws Exception {
-    runServer(Collections.emptySet(), Set.of(new CorsFilter()));
+    getServer(new ServerFactory.Api()).start();
```

Comment:
should we split up `getServer`? 

reads like two separate methods now:
1) `verifyAndMigrate`
2) `startServer`

thinking out loud.

Reply:
Agreed, but there is sequential object creation throughout `verifyAndMigrate` where we'd need to return a context (?) that contains the various objects we're instantiating, and consuming that in `startServer`?

Reply:
Agreed, but there is sequential object creation throughout `verifyAndMigrate` where we'd need to return a context (?) that contains the various objects we're instantiating, and consuming that in `startServer`?